### ISSUE TITLE
Throttling is not working correctly if promise chain is using route method

### ISSUE THREAD

**Issue by pojaszek** [ratpack/ratpack#1281]
I think I found a bug in ratpack promises. There is a problem when I use combination of route and throttled method.
```
    def "bug with route and throttled"() {
        given:
        Throttle concurrentExecutions = Throttle.ofSize(10)
        List<Promise> promises = []

        for (int i = 0; i < 100; i++) {
            promises.add(Promise.value(i)
                .route({ it > 10}, {println("Route value ${it}")})
                .throttled(concurrentExecutions))
        }
        when:
        ExecHarness.executeSingle {
            ParallelBatch.of(promises).yieldAll().operation()
        }

        then:
        1 != 1
    }
```
Execution result of above unit test is:
```
Route value 21
Route value 13
Route value 12
Route value 16
Route value 17
Route value 11
Route value 18
Route value 14
Route value 19
Route value 20
```
and then the execution is suspended.

I reckon that route operation doesn't remove lock from throttling stage.

**Comment by ldaley** [comment]
Thanks for the report. Will be fixed in 1.5.1.

**Comment by ldaley** [comment]
Fixed by https://github.com/ratpack/ratpack/commit/be4116679235de1f129f421b59d1671556d8d635
